src/cephadm/cephadmlib/node_proxy/redfishdellsystem.py

```diff
@@ -0,0 +1,186 @@
+import json
+from .baseredfishsystem import BaseRedfishSystem
+from .util import Logger, normalize_dict, to_snake_case
+from typing import Dict, Any, List
+
+
+class RedfishDellSystem(BaseRedfishSystem):
+    def __init__(self, **kw: Any) -> None:
+        super().__init__(**kw)
+        self.log = Logger(__name__)
+
+    def build_common_data(self,
+                          data: Dict[str, Any],
+                          fields: List,
+                          path: str) -> Dict[str, Dict[str, Dict]]:
+        result: Dict[str, Dict[str, Dict]] = dict()
+        for member_info in self.get_members(data, path):
+            member_id = member_info['Id']
+            result[member_id] = dict()
+            for field in fields:
+                try:
+                    result[member_id][to_snake_case(field)] = member_info[field]
+                except KeyError:
+                    self.log.logger.warning(f'Could not find field: {field} in member_info: {member_info}')
```

Comment:
I'm a bit confused by this. It looks like `result[member_id]` is initialized to an empty dict, but then there's handling for if it doesn't have some field in it. Wouldn't it have no fields yet since this loop over the fields is the first thing done since initializing to an empty dict?

Reply:
See a similar pattern in some other `build_<thing>` funcs

Reply:
> there's handling for if it doesn't have some field in it

that handling is for the case where `field` is not present in `member_info`

Reply:
my mistake, I thought the key error check was for a key error on `result` at first, but it's actually for `member_info` and makes sense